### DISCUSSION TITLE
Strange osu! classic skin fallback behavior

### DISCUSSION THREAD

**Discussion by Yamakyu** [ppy/osu#13432] [Q&A] [answered]
Greetings,

I open this discussion in regards to a strange skin behavior that started happening upon Lazer 2021.608.0 update. 

I use a custom skin, imported by using the "import from stable" feature of Lazer. The main purpose of this skin is to **remove approach circles** (as well as change a couple less important things, like removing feedback for 300/great hits), and for that reason it's a very incomplete skin that relies mostly on osu! replacing missing elements (i.e. most elements) with default assets. I acknowledge that 1/ my skin is susceptible to working unproperly since it's lacking so many elements (it has worked well so far) 2/ my knowledge of skinning is limited, and could ultimately be the source of my problems here.

My skin was working mostly very well in Lazer, until Lazer 2021.608.0 update, when Lazer started falling back missing assets to classic elements instead of Lazer elements (as it used to be until then). I believe this is what's implied in 

> **Allow fallback to legacy defaults from legacy skins (osu#13264)par peppy**
> User skins (aka "legacy" skins) will now correctly get osu!classic resources if they aren't present in the user skin itself. This fixes cases where users expect certain textures / elements to display but were instead missing.

Correct ?

Aside from the change in default assets, I have 2 concerns with this change. 

1. The asset used to provide score feedback (i.e. 300/100/50 or Great/Good/Meh) stays on screen for much longer upon hitting circles. I do not have footage of how it used to be sadly, but it is clear to me that these remain displayed for a much longer time than before. However after further investigation I suspect that it has to do with my lack of knowledge of skinning, and I feel like I might be able to do something about it. 
2. _My main concern_ however is that [combos of 10 or more started glitching out of their circles](https://imgur.com/a/vIcgPS6). It wasn't the case before in Lazer, and this started happening in update 2021.608.0. This also happens in osu!Classic.

![vlcsnap-2021-06-10-13h26m31s975](https://user-images.githubusercontent.com/43894102/121517924-2f6def80-c9f0-11eb-8cf0-51fd00db8122.png)

I actually have encountered this behavior long ago in osu!Classic when I was using another custom skin without approach circles (which I sadly don't have anymore), but it had more way elements than the one I'm using right now.
I believe this has to do with the (many) missing elements of my skin, and as I said I know my skin knowledge is limited. But it worked very well back when Lazer was using Lazer elements by default instead of Classic elements.

I would like to know how I can work around this, or if it is possible to tell Lazer to use Lazer skin elements by default, since it worked so well. 

[Attached here](https://bsolife.fr/public/YamaK/Yamakyu_Skin-Issues_-_Lazer_2021.608.0.zip
) is :

- Video footage. I recorded 2 replays in which this glitch happens multiple times. It's very common in [this beatmap](https://osu.ppy.sh/beatmapsets/1232158#osu/2561468) I created (it's not as refined as a proper ranked beatmap, and has many issues including long combos), and just to be sure I checked in other actually ranked beatmaps -[like this one](https://osu.ppy.sh/beatmapsets/171388#osu/671750), included in the .zip- and it happens still.
- My skin, for reviewing and testing if necessary. Also, because Lazer now uses Classic elements by default I actually added .png files for Good/Meh feedbacks, as well as the cursor and cursortrail after the update, so that I don't use the default classic ones. Everything else was already there before.

That is all. 
Good day :>

**Comment by peppy** [comment, marked as answer]
Would appreciate if someone can sum this up into a sentence or two.

**Reply by bdach** [reply]
I *think* the executive summary is: broken skin. User was relying on previous fallback behaviour of legacy skins to default lazer skin and specified a `HitCircleOverlap` of 180. Adding back the proper legacy fallback logic broke this skin as it was relying on unsupported behaviour.

**Reply by Yamakyu** [reply]
Sorry for the long post, thought it would be better that way.

From your answer I assume this issue is mostly my doing ?

**Reply by bdach** [reply]
To fix the numbers showing like they currently are you need to edit the skin to provide a saner value for `HitCircleOverlap`, yes.

**Reply by Yamakyu** [reply]
Thank you very much for your help
